packages/nodes-base/credentials/Aws.credentials.ts

```diff
@@ -275,6 +327,127 @@ export class Aws implements ICredentialType {
 				password: true,
 			},
 		},
+		{
+			displayName: 'Role ARN',
+			name: 'roleArn',
+			description: 'The ARN of the role to assume (e.g., arn:aws:iam::123456789012:role/MyRole)',
+			type: 'string',
+			displayOptions: {
+				show: {
+					credentialType: ['assumeRole'],
+				},
+			},
+			required: true,
+			default: '',
+			placeholder: 'arn:aws:iam::123456789012:role/MyRole',
+		},
+		{
+			displayName: 'External ID',
+			name: 'externalId',
+			description:
+				"External ID for cross-account role assumption (should be required by your role's trust policy). This really should be generated automatically by n8n but there is not a way to do this automatically. Given this limitation, **you should treat this value as a secret and not share it with any other users of this n8n instance.** For more information, see https://docs.aws.amazon.com/IAM/latest/UserGuide/confused-deputy.html",
+			type: 'string',
+			displayOptions: {
+				show: {
+					credentialType: ['assumeRole'],
+				},
+			},
+			required: true,
+			default: '',
+			typeOptions: {
+				password: true,
+			},
+		},
+		{
+			displayName: 'Role Session Name',
+			name: 'roleSessionName',
+			description: 'Name for the role session (required, defaults to n8n-session)',
+			type: 'string',
+			displayOptions: {
+				show: {
+					credentialType: ['assumeRole'],
+				},
+			},
+			required: true,
+			default: 'n8n-session',
+		},
+		{
+			displayName: 'Use System Credentials for STS Call',
+			name: 'useSystemCredentialsForRole',
+			description:
+				'Use system credentials (environment variables, container role, etc.) to call STS.AssumeRole. If system credentials are not configured by your administrator, you must provide an access key id and secret access key below that has the necessary permissions to assume the role.',
+			type: 'boolean',
+			displayOptions: {
+				show: {
+					credentialType: ['assumeRole'],
```

Comment:
Why only use system credential for `assumeRole`? Sounds like it should just be a global option alternative 

* **Either** you use explicit API keys **or** you let AWS SDK load credentials from the system.
* Then **Either** you use assume a new role **or** keep the current one?

Like, why have `STS Access Key ID` AND `Access Key ID`?

Reply:
@isra17 because there needs to be an alternative to using the system credentials to assume a role if the n8n admin has not configured them. STS Access Key ID has a different role than Access Key ID and is a credential provided by the n8n user that is only used to assume a role and do nothing else. This follows best practice of role assumption for IAM Users.

Reply:
@isra17 also, we can't use the system credential permissions for workflows. This would allow the n8n user to have permissions to access items within the AWS account that n8n is running in, in the case that the system credentials are coming from an IAM Role attached to the n8n deployment environment.

Reply:
Why does it matter wheter these credential are used by the workflow or not? You semantic distinction duplicate many fields in the option datatype where half the duplicated fields are not used based on the `credentialType`. STS Access Key are simply AWS Access Key. In this case, the workflow just happen to get a different set of credential because an AssumeRole step was added.

At the end of the day, it should be as simple as "Plug in your AWS credentials in these fields, and if you want to assume role, just add it to the role field". There's no reason to make it more complicated than that, both in the code, the UI and in the doc.